File: L9/logo.cpp
#define _USE_MATH_DEFINES
#include <iostream>
#include <math.h>

using namespace std;

int dist(double x, double y) {
    double distance = sqrt(pow(x, 2) + pow(y, 2));
    return round(distance);
}

int main()
{
    int ncases;
    cin >> ncases;
    while (ncases--)
    {
        int ncommands;
        cin >> ncommands;
        double currx = 0, curry = 0;
        int angle = 0; 
        while (ncommands--)
        {
            string dir;
            int dist;
            cin >> dir >> dist;
            if (dir == "fd")
            {
                currx += dist * sin(angle * (M_PI / 180));
                curry += dist * cos(angle * (M_PI / 180));
            }
            else if (dir == "bk")
            {
                currx -= dist * sin(angle * (M_PI / 180));
                curry -= dist * cos(angle * (M_PI / 180));
            }
            else if (dir == "lt")
                angle += dist;
            else
                angle -= dist;
        }
        printf("%d\n", dist(currx, curry));
    }
    return 0;
}

File: L9/logo.py
from math import pi, cos, sin

cases = int(input())

for _ in range(cases):
    commands = int(input())
    x = 0.
    y = 0.
    angle = 0
    for _ in range(commands):
        direction, distance = input().split()
        distance = int(distance)
        if (direction == "fd"):
            x += distance * sin(angle * (pi/180))
            y += distance * cos(angle * (pi/180))
        elif (direction == "bk"):
            x -= distance * sin(angle * (pi/180))
            y -= distance * cos(angle * (pi/180))
        elif (direction == "rt"):
            angle -= distance
        else:
            angle += distance
    distance = round((x**2 + y**2)**0.5)
    print(distance)